netty/src/main/java/io/grpc/netty/WriteBufferingAndExceptionHandler.java

```diff
@@ -139,6 +150,22 @@ public void operationComplete(ChannelFuture future) {
     promise.addListener(new ConnectListener());
   }
 
+  @Override
+  public void channelRead(ChannelHandlerContext ctx, Object msg) {
+    if (logger.isLoggable(Level.FINE)) {
+      Object loggedMsg = msg instanceof ByteBuf ? ByteBufUtil.hexDump((ByteBuf) msg) : msg;
+      logger.log(
+          Level.FINE,
+          "Unexpected read {0} reached end of pipeline {1}",
+          new Object[] {loggedMsg, ctx.pipeline().names()});
+    }
+    ReferenceCountUtil.safeRelease(msg);
+    exceptionCaught(
+        ctx,
+        Status.INTERNAL.withDescription("read() missed by ProtocolNegotiator handler")
```

Comment:
I'm sort of questioning why we don't include the bytes directly in this message instead of logging them. But it does log the entire message, so I can sort of understand.

If we aren't going to include the payload, how about we include the msg type in the status? That's generally pretty useful.

Reply:
I added the toString() repr, which for bytebufs should be the length and type.  The long message is what I wanted to avoid.